### PR TITLE
w263768889 Show EB Deployments in Rollbar

### PR DESCRIPTION
**Pull request by jeet4320**
Closes https://www.wrike.com/open.htm?id=263768889

## What
Dependent on this [PR](https://github.com/bodyshopbidsdotcom/PyRollbarAPI/pull/1)

Tested this PR on project `snapsheet-url-shortener` https://rollbar.com/snapsheettech/snapsheet-url-shortener/deploys/

More changes:
1. If PR already has qa label before deployment and if deployment fails then script will remove Pushing label if it is present on open PR.
2. If PR has no qa label before deployment and if deployment fails then script will remove qa and Pushing label if those labels are on open PR.
3. report_deploy_to_rollbar will be called if and only if deployment passes.
4. Throw info message if rollbar project is not present

## Testing

Run `git_push_eb qa2` by commenting build, push, deploy functions.

### CODE REVIEW

**Comment by jeet4320** [comment]
The PR failed test is expected as `git-scripts` repo doesn't have access to private `PyRollbarAPI` repo

File: requirements.txt [hunk 1]
```diff
@@ -6,9 +6,9 @@ smmap==0.9.0
...
 githubgateway==0.8
-PyRollbarAPI==0.6
 PyGithub==1.26.0
 inflection==0.3.1
 PyYAML==3.11
 awscli==1.15.63
-awsebcli==3.14.3
\ No newline at end of file
+awsebcli==3.14.3
+-e git+git@github.com:bodyshopbidsdotcom/PyRollbarAPI.git@w267070023-add-api-to-create-deploy-scrip#egg=PyRollbarAPI
```
Comment (santi-h) [inline requirements.txt]: Let me make a new release for `PyRollbarAPI` so we don't have to do this

Comment (whatisdot) [reply requirements.txt]: This was so we could use this as a test. After pyrollbar is merged we should update the version.

**Comment by jeet4320** [comment]
@whatisdot you wont see that message now if you run `pip install -r requirements.txt` after `workon git-scripts` due to this change https://github.com/bodyshopbidsdotcom/PyRollbarAPI/pull/1/commits/d3d7f45c79abbc992bfae18da7d56ddb8ef0d06e

**Comment by whatisdot** [comment]
@jeet4320 Now when I run I see this:
```
9b2e-qa5-web - INFO: Environment update completed successfully.

Alert: An update to the EB CLI is available. Run "pip install --upgrade awsebcli" to get the latest version.
[2018-08-30 14:12:08,005]: global name 'GithubAPIGateway' is not defined
```

**Comment by jeet4320** [comment]
@whatisdot sorry about this.. I have added imports now and tested it once.

**Comment by whatisdot** [comment]
I see the logged lines that say it's connecting to Rollbar...
```bash
```

...but I'm not seeing deployments show up: https://rollbar.com/snapsheettech/claim_system_backend/deploys/

Could this be because the git project is named "vice-backend" and Rollbar is reporting it as "claim_system_backend"?

**Comment by jeet4320** [comment]
@whatisdot yes you are right. I create instance `PyRollbarAPI` with current repo https://github.com/bodyshopbidsdotcom/git-scripts/pull/372/files#diff-cf66ab1114b91c1322d5cbcefc7fbcb7R161

File: lib/push_eb.py [hunk 2]
```diff
@@ -142,14 +144,23 @@ def command(self):
...
     self.config.cleanup()
     self.logger.info('Push complete.')
     return ['']
 
+  def report_deploy_to_rollbar(self, environment):
+    helper2 = Helper2()
+    owner, repo = helper2.owner_and_repo()
+    github_api = GithubAPIGateway(*helper2.owner_and_repo(), token=Helper2.get_var('GITHUB_TOKEN'))
+    current_branch = helper2.current_branch()
+    remote_sha = helper2.remote_head_sha(dst_branch = current_branch)
```
Comment (whatisdot) [inline lib/push_eb.py]: This should be the `local_sha`.

**Comment by jeet4320** [comment]
I think `vice-backend` rollbar project name issue should not block this PR to be merged

**Comment by whatisdot** [comment]
I ran `git_push_eb stage` on `snap-hub-api`, and I saw this at the end of the log:
```
[2018-09-06 10:12:11,789]: Starting new HTTPS connection (1): api.rollbar.com
[2018-09-06 10:12:11,853]: Starting new HTTPS connection (1): api.rollbar.com
[2018-09-06 10:12:11,911]: ('Connection aborted.', error(22, 'Invalid argument'))
```

**Comment by whatisdot** [comment]
Also, what is the behavior if the EB part of the deploy fails? Like, if we try to deploy while an update is already in progress?

**Comment by jeet4320** [comment]
@whatisdot `snap-hub-api` is not present on rollbar. On rollbar it is [snap-hub](https://rollbar.com/snapsheettech/snap-hub/items)

**Comment by jeet4320** [comment]
@whatisdot I have added new changes. 

1. If PR has `qa` label before deployment and if deployment fails then script will remove `Pushing` label if it is present on open PR.
2. If PR has no `qa` label before deployment and if deployment fails then script will remove `qa` and `Pushing` label if those labels are on open PR.
3. Also `report_deploy_to_rollbar` will be called if and only if deployment passes.

**Comment by jeet4320** [comment]
@nickstanish @santi-h  `snap-hub-api` project has rollbar project as `snap-hub`, can you rename rollbar project to match it with github repo name. This PR report deployments to rollbar by using git repo name. So git repo name and rollbar project name should match with each other.

**Comment by santi-h** [comment]
@jeet4320 I'll update it

**Comment by santi-h** [comment]
@jeet4320 done

**Comment by jeet4320** [comment]
@santi-h can you approve the PR as well :)

**Pull request merged**